slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt

```diff
@@ -436,82 +440,88 @@ class SlideToActView @JvmOverloads constructor (
+            finalPositionAnimator.addUpdateListener {
+                mPosition = it.animatedValue as Int
+                invalidateArea()
             }
-            invalidateArea()
-        }
 
-        val tickAnimator: ValueAnimator
-        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N) {
-            // Fallback not using AVD.
-            tickAnimator = ValueAnimator.ofInt(0, 255)
-            tickAnimator.addUpdateListener {
-                mTickMargin = mIconMargin
-                mFlagDrawTick = true
-                mDrawableTick.alpha = it.animatedValue as Int
+            // Animator that bounce away the cursors
+            val marginAnimator = ValueAnimator.ofInt(mActualAreaMargin, (mInnerRect.width() / 2).toInt() + mActualAreaMargin)
+            marginAnimator.addUpdateListener {
+                mActualAreaMargin = it.animatedValue as Int
                 invalidateArea()
             }
-        } else {
-            // Used AVD Animation.
-            tickAnimator = ValueAnimator.ofInt(0)
-            tickAnimator.addUpdateListener {
-                if (!mFlagDrawTick) {
+            marginAnimator.interpolator = AnticipateOvershootInterpolator(2f)
+
+            // Animator that reduces the outer area (to right)
+            val areaAnimator = ValueAnimator.ofInt(0, (mAreaWidth - mAreaHeight) / 2)
+            areaAnimator.addUpdateListener {
+                mActualAreaWidth = it.animatedValue as Int
+                if (Build.VERSION.SDK_INT >= 21) {
+                    invalidateOutline()
+                }
+                invalidateArea()
+            }
+
+            val tickAnimator: ValueAnimator
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N) {
+                // Fallback not using AVD.
+                tickAnimator = ValueAnimator.ofInt(0, 255)
+                tickAnimator.addUpdateListener {
                     mTickMargin = mIconMargin
                     mFlagDrawTick = true
-                    startTickAnimation()
+                    mDrawableTick.alpha = it.animatedValue as Int
                     invalidateArea()
                 }
+            } else {
+                // Used AVD Animation.
+                tickAnimator = ValueAnimator.ofInt(0)
+                tickAnimator.addUpdateListener {
+                    if (!mFlagDrawTick) {
+                        mTickMargin = mIconMargin
+                        mFlagDrawTick = true
+                        startTickAnimation()
+                        invalidateArea()
+                    }
+                }
             }
-        }
 
-        if (mPosition >= mAreaWidth - mAreaHeight) {
-            animSet.playSequentially(marginAnimator, areaAnimator, tickAnimator)
-        } else {
-            animSet.playSequentially(finalPositionAnimator, marginAnimator, areaAnimator, tickAnimator)
-        }
+            if (mPosition >= mAreaWidth - mAreaHeight) {
```

Comment:
What do you think about something like this:

```kotlin
val animators = mutableListOf<Animator>()
if (mPosition < mAreaWidth - mAreaHeight) {
    animators.add(finalPositionAnimator)
}
if (isAnimateComplete){
    animators.add(marginAnimator)
    animators.add(areaAnimator)
    animators.add(tickAnimator)
}
animSet.playSequentially()
```

And for the `startResetAnimation` you probably need something like this:
```
if (isAnimateComplete) {
    animSet.playSequentially(tickAnimator, areaAnimator, positionAnimator, marginAnimator, arrowAnimator)
} else {
    animSet.playSequentially(positionAnimator)
}
```

Reply:
Done! :)